receiver/statsdreceiver/receiver.go

```diff
@@ -69,12 +69,13 @@ func New(
 
 func buildTransportServer(config Config) (transport.Server, error) {
 	// TODO: Add TCP/unix socket transport implementations
-	switch strings.ToLower(config.NetAddr.Transport) {
-	case "", "udp":
-		return transport.NewUDPServer(config.NetAddr.Endpoint)
+	trans, err := transport.NewTransport(strings.ToLower(config.NetAddr.Transport))
```

Comment:
isn't this error unmanaged? Should we check it first before leveraging the other with `switch trans`?

We are assuming that trans is correct but it could be not

Reply:
I removed `NewTransport` to return an error so there is no unmanaged error here.

The error I return now is the same already was on the receiver so there is no change in the behavior/error returned.